### PR TITLE
Add pyspark support

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

Add pyspark support

### CODE REVIEW

File: extensions/spark/setup/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env python
+#
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (frankfliu) [inline extensions/spark/setup/__init__.py:3]:
```suggestion
# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

File: extensions/spark/setup/djl_spark/__init__.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env python
+#
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (frankfliu) [inline extensions/spark/setup/djl_spark/__init__.py:3]:
```suggestion
# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

File: extensions/spark/setup/djl_spark/transformer.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+#
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (frankfliu) [inline extensions/spark/setup/djl_spark/transformer.py:3]:
```suggestion
# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

File: extensions/spark/setup/djl_spark/translator.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+#
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (frankfliu) [inline extensions/spark/setup/djl_spark/translator.py:3]:
```suggestion
# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

File: docker/spark/Dockerfile [hunk 5]
```diff
@@ -24,6 +24,8 @@ ADD https://repo1.maven.org/maven2/ai/djl/onnxruntime/onnxruntime-engine/${DJL_V
 ADD https://repo1.maven.org/maven2/ai/djl/tensorflow/tensorflow-engine/${DJL_VERSION}/tensorflow-engine-${DJL_VERSION}.jar /usr/lib/spark/jars/djl-tensorflow-engine-${DJL_VERSION}.jar
 ADD https://repo1.maven.org/maven2/net/java/dev/jna/jna/${JNA_VERSION}/jna-${JNA_VERSION}.jar /usr/lib/spark/jars/
 RUN chmod -R +r /usr/lib/spark/jars/
+## TODO: Update url
+RUN pip3 install --no-cache-dir https://raw.githubusercontent.com/xyang16/djl/master/djl_spark-${DJL_VERSION}-py3-none-any.whl
```
Comment (frankfliu) [inline docker/spark/Dockerfile:28]: Can we build the wheel during dockder build, or build the wheel before docker build?

Comment (xyang16) [reply docker/spark/Dockerfile:28]: To build the wheel during docker build, we need pull the djl repo. Is this OK.

File: extensions/spark/setup/__init__.py (hunk 1 above)
Comment (xyang16) [reply extensions/spark/setup/__init__.py]: Done

File: extensions/spark/setup/djl_spark/__init__.py (hunk 2 above)
Comment (xyang16) [reply extensions/spark/setup/djl_spark/__init__.py]: Done

File: extensions/spark/setup/djl_spark/transformer.py (hunk 3 above)
Comment (xyang16) [reply extensions/spark/setup/djl_spark/transformer.py]: Done

File: extensions/spark/setup/djl_spark/translator.py (hunk 4 above)
Comment (xyang16) [reply extensions/spark/setup/djl_spark/translator.py]: Done

File: extensions/spark/README.md [hunk 6]
```diff
@@ -23,3 +37,38 @@ You can pull the module from the central Maven repository by including the follo
...
+
+```python
+from djl_spark.transformer import SparkTransformer
+from djl_spark.translator import SparkImageClassificationTranslator
+
+transformer = SparkTransformer(input_col="input_col",
+                               output_col="value",
+                               model_url="model_url",
+                               output_class="ai.djl.modality.Classifications",
+                               translator=SparkImageClassificationTranslator())
```
Comment (lanking520) [inline extensions/spark/README.md:70]: We may also need a Python generic Python Translator. User can implements based on that. Inside the python translator, we embedded the numpy translator in Java. Where we accept a numpy array input and return as numpy array outputs.

File: docker/spark/Dockerfile (hunk 5 above)
Comment (xyang16) [reply docker/spark/Dockerfile]: Build the wheel before docker build.

**Review by lanking520 — APPROVED** [review]

**Pull request merged**